From 2d1a3062bcb0e0716168c75b9c4ccef17189a47d Mon Sep 17 00:00:00 2001
From: Peter Dons Tychsen <pdt@dontech.dk>
Date: Sun, 21 Jul 2019 23:38:02 +0200
Subject: [PATCH] msvcrt: Do not use casting in min/max macros

Macros like ULLONG_MAX are meant to be able to be used in macro conditionals like "#if (UULONG_MAX >= 1)". This is not possible as there are casts in them currently which of course the preprocessor cannot parse.

Fix this by using type prefixes instead which the preprocessor does understand.
This makes libraries like "libtom" compile with Wine headers.

Signed-off-by: Peter Dons Tychsen <pdt@dontech.dk>
---
 include/msvcrt/limits.h | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/include/msvcrt/limits.h b/include/msvcrt/limits.h
index 13bab8679ba..8d3a0387cd3 100644
--- a/include/msvcrt/limits.h
+++ b/include/msvcrt/limits.h
@@ -30,13 +30,13 @@
 #define LONG_MAX    0x7fffffffL
 #define ULONG_MAX   0xffffffffUL
 
-#define LLONG_MAX   (((__int64)0x7fffffff << 32) | 0xffffffff)
+#define LLONG_MAX   ((0x7fffffffLL << 32) | 0xffffffff)
 #define LLONG_MIN   (-LLONG_MAX-1)
-#define ULLONG_MAX  (((unsigned __int64)0xffffffff << 32) | 0xffffffff)
+#define ULLONG_MAX  ((0xffffffffULL << 32) | 0xffffffff)
 
-#define _I64_MAX    (((__int64)0x7fffffff << 32) | 0xffffffff)
+#define _I64_MAX    ((0x7fffffffLL << 32) | 0xffffffff)
 #define _I64_MIN    (-_I64_MAX-1)
-#define _UI64_MAX   (((unsigned __int64)0xffffffff << 32) | 0xffffffff)
+#define _UI64_MAX   ((0xffffffffULL << 32) | 0xffffffff)
 
 #define I64_MIN  _I64_MIN
 #define I64_MAX  _I64_MAX
-- 
2.17.1

